### PR TITLE
ci: automatically update emqx/emqx:latest on releases

### PR DESCRIPTION
**Pull request by id**
Fixes EMQX-8284

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/` dir
- [ ] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in

### CODE REVIEW

File: .github/workflows/build_and_push_docker_images.yaml [hunk 1]
```diff
@@ -41,9 +44,14 @@ jobs:
           cd source
           tag=${{ github.ref }}
           # tag docker-latest-ce or docker-latest-ee
-          if git describe --tags --exact --match 'docker-latest-*'; then
+          if git describe --tags --exact --match 'docker-latest-*' 2>/dev/null; then
```
Comment (qzhuyan) [inline .github/workflows/build_and_push_docker_images.yaml:47]: the latest tag is on 4.4? or 5.0?

File: .github/workflows/build_and_push_docker_images.yaml [hunk 2]
```diff
@@ -41,9 +44,14 @@ jobs:
...
+            echo 'docker_latest=true due to docker-latest-* tag'
```
Comment (thalesmg) [inline .github/workflows/build_and_push_docker_images.yaml:48]: For my own information: is this tagged automatically somehow?  I couldn't find it in any workflows. :thinking:

**Review by thalesmg — APPROVED** [review]

File: .github/workflows/build_and_push_docker_images.yaml (hunk 1 above)
Comment (id) [reply .github/workflows/build_and_push_docker_images.yaml:47]: it's on 5.0.11 now for both emqx/emqx and emqx

File: .github/workflows/build_and_push_docker_images.yaml (hunk 2 above)
Comment (id) [reply .github/workflows/build_and_push_docker_images.yaml:48]: It's not documented anywhere, does not exist in any process checklist and not automated, hence the PR.
I left an option to use `docker-latest*` to override releases if needed, but in general `:latest` should be automatically assigned on new v5.x releases.

**Pull request merged**